src/kaocha/plugin/junit_xml.clj

```diff
@@ -100,32 +100,35 @@
                         (hierarchy/fail-type? m) (failure->xml m)))
                     events)}))
 
-(defn suite->xml [suite index]
+(defn suite->xml [{:keys [no-system-out?]} suite index]
   (let [id (::testable/id suite)]
-    {:tag :testsuite
-     :attrs (-> {:name (test-name suite)
-                 :id index
-                 :hostname "localhost"}
-                (merge (stats suite) (time-stat suite))
-                (assoc :package (if (qualified-ident? id)
-                                  (namespace id)
-                                  "")))
+    {:tag     :testsuite
+     :attrs   (-> {:name     (test-name suite)
+                   :id       index
+                   :hostname "localhost"}
+                  (merge (stats suite) (time-stat suite))
+                  (assoc :package (if (qualified-ident? id)
+                                    (namespace id)
+                                    "")))
      :content (concat
-               [{:tag :properties}]
-               (map testcase->xml (leaf-tests suite))
-               [{:tag :system-out
-                 :content (->> suite
-                               test-seq
-                               (keep :kaocha.plugin.capture-output/output)
-                               (remove #{""})
-                               (map strip-ansi-sequences))}
-                {:tag :system-err}])}))
+                [{:tag :properties}]
+                (map testcase->xml (leaf-tests suite))
+                [(merge {:tag :system-out}
+                        (when-not no-system-out?
+                          {:content (->> suite
+                                         test-seq
+                                         (keep :kaocha.plugin.capture-output/output)
```

Comment:
Just realized this but I think you could have also disabled the output capturing plugin on CI to get the same result.

Reply:
Not really. Disabling output capturing plugin will print out stdout immediately. It won't be stored in the XML but it will be saved in the build log